s3/src/main/scala/akka/stream/alpakka/s3/auth/CanonicalRequest.scala

```diff
@@ -59,7 +59,13 @@ private[alpakka] object CanonicalRequest {
       pathEncodeRec(builder ++= uriEncodePath(head), tail)
   }
 
-  private def toHexUtf8(ch: Char): String = "%" + Integer.toHexString(ch.toInt)
+  private def toHexUtf8(ch: Char): String = ch match {
+    case asciiChar if ch >= 0 && ch <= 127 =>
+      "%" + Integer.toHexString(asciiChar.toInt).toUpperCase
+    case otherChar =>
+      // I guess this adds quite some overhead but the detailed api is not public
+      Path(otherChar.toString).toString()
```

Comment:
Personally not too fond of this part but I want to avoid re-inventing the wheel on uri encoding